### DISCUSSION TITLE
Getting pose priors from blender

### DISCUSSION THREAD

**Discussion by UkuKert** [openMVG/openMVG#2057] [General]
Hello! First of all I want to thank you guys for this awesome tool. It is truly amazing, what smart people can accomplish when they come together!

I have been using this tool for my Bachelors thesis where I used it to regenerate a comet nucleus based on images taken by "fake" satellites I simulated in blender. For that I used ComputeStructureFromKnownPoses with the images and poses exported from blender. For the past week it has been a headache, because the definitions of the rotation matrix `R` are different in blender and openMVG. I tried using several tools for debugging purposes, but the problem didn't show itself. I discovered the solution by sheer luck and I want to share the transformation for anyone else struggling in the future.

In order to transform the Blender rotation matrix `R` to OpenMVG rotation matrix `R` you have to do the following:
```
import numpy as np
#R_blender is a Rotation matrix exported from blender
#C_blender is a position exported from blender

R_conv_mat = np.array([[1,  0,  0], 
                       [0, -1,  0], 
                       [0,  0, -1]])
R_openMVG = np.matmul(R_blender,R_conv_mat).T.tolist()
C_openMVG = C_blender
```
Keep in mind that in blender the up axis is z and in openMVG it is y so when you export the model back to blender, you have to rotate around the x axis -90deg. Or rotate the `C` and `R` matrixes, but since I haven't tested it, I will not add that part here.

I hope this helps someone in the future.

**Comment by pmoulon** [comment]
Thank you for your message to the community!
